File: 268-missing-number/268-missing-number.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var missingNumber = function(nums) {
    let numCheck = new Set(nums)
    for (let i = 0; i < nums.length; i++) {
        if (numCheck.has(i)) {
            continue
        } else {
            return i
        }
    }
    return nums.length
};

File: 268-missing-number/268-missing-number.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        int len = std::size(nums);
        std::set<int> numCheck(std::begin(nums), std::end(nums));
        for(int i = 0; i < len; ++i)
        {
            if (numCheck.count(i)) {
                continue;
            }
            else
            {
                return i;
            }
        }
        return len;
    }
};